packit_service/models.py

```diff
@@ -356,6 +357,45 @@ def __str__(self):
         return self.__repr__()
 
 
+class TaskResult(Task):
+    @classmethod
+    def get_by_id(cls, task_id: str) -> Optional["TaskResult"]:
+        with get_sa_session() as session:
+            return session.query(TaskResult).filter_by(task_id=task_id).first()
+
+    @classmethod
+    def get_all(cls) -> Optional[Iterable["TaskResult"]]:
+        with get_sa_session() as session:
+            return session.query(TaskResult).all()
+
+    # needed in migration from redis to psql and used in tests
+    @classmethod
+    def add_task_result(cls, task_id, status, result, traceback, date_done):
+        with get_sa_session() as session:
+            task_result = cls.get_by_id(task_id)
+            if task_result is not None:
+                task_result.status = status
+                task_result.result = result
+                task_result.traceback = traceback
+                task_result.date_done = date_done
+                session.add(task_result)
+                return task_result
+            else:
+                task_result = cls(task_id)
+                task_result.status = status
+                task_result.result = result
+                task_result.traceback = traceback
+                task_result.date_done = date_done
+                session.add(task_result)
+                return task_result
```

Comment:
I know it's still WIP, but small improvement:

```suggestion
            if task_result is None:
                task_result = cls(task_id)
            task_result.status = status
            task_result.result = result
            task_result.traceback = traceback
            task_result.date_done = date_done
            session.add(task_result)
            return task_result
```